provider/info/info_data_source.go

```diff
@@ -0,0 +1,151 @@
+				Description: "OCM account User full name",
+				Computed:    true,
+			},
+			"account_username": schema.StringAttribute{
+				Description: "OCM account username",
+				Computed:    true,
+			},
+			"account_email": schema.StringAttribute{
+				Description: "OCM account email",
+				Computed:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "OCM account organization id",
+				Computed:    true,
+			},
+			"organization_external_id": schema.StringAttribute{
+				Description: "OCM account organization external id",
+				Computed:    true,
+			},
+			"organization_name": schema.StringAttribute{
+				Description: "OCM account organization name",
+				Computed:    true,
+			},
+			"ocm_api": schema.StringAttribute{
+				Description: "OCM API url",
+				Computed:    true,
+			},
+			"ocm_aws_account_id": schema.StringAttribute{
+				Description: "OCM AWS account ID",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (d *OCMInfoDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured:
+	if req.ProviderData == nil {
+		return
+	}
+
+	// Cast the provider data to the specific implementation:
+	connection := req.ProviderData.(*sdk.Connection)
+
+	// Get the collection of cloud providers:
+	d.collection = connection.AccountsMgmt().V1().CurrentAccount()
+	d.ocmAPI = connection.URL()
+
+}
+
+func (d *OCMInfoDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	// Get the state:
+	state := &OCMInfoState{}
+	diags := req.Config.Get(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Get account information
+	accountResp, err := d.collection.Get().Send()
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to connect to OCM", "Please verify your OCM offline token")
+		return
+	}
+	obj, ok := accountResp.GetBody()
+	if !ok {
+		resp.Diagnostics.AddError("Parsing Error", "Failed to parse OCM response")
+		return
+	}
+
+	state.AccountID = types.StringValue(obj.ID())
+	state.AccountName = types.StringValue(fmt.Sprintf("%s %s", obj.FirstName(), obj.LastName()))
+	state.AccountUsername = types.StringValue(obj.Username())
+	state.AccountEmail = types.StringValue(obj.Email())
+	state.OrganizationID = types.StringValue(obj.Organization().ID())
+	state.OrganizationExternalID = types.StringValue(obj.Organization().ExternalID())
+	state.OrganizationName = types.StringValue(obj.Organization().Name())
+
+	state.OCMAPI = types.StringValue(d.ocmAPI)
+	state.OCMAWSAccountID = types.StringValue(extractOCMAWSAccount(d.ocmAPI))
+
+	// Save the state:
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func extractOCMAWSAccount(url string) string {
+	env := ocmEnvProd
+	if strings.Contains(url, "stage") {
+		env = ocmEnvStage
+	} else if strings.Contains(url, "integration") {
+		env = ocmEnvInt
+	}
```

Comment:
URL (connection.URL()) will always return a value, right?

Reply:
yes, its part of the provider (client) setup